numpy/core/src/umath/override.c

```diff
@@ -576,19 +583,18 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
             break;
         }
     }
-
+    status = 0;
     /* Override found, return it. */
-    Py_XDECREF(method_name);
-    Py_XDECREF(normal_kwds);
-    Py_DECREF(override_args);
-    return 0;
-
+    goto cleanup;
```

Comment:
I had started with that but then realised there are 10 `goto fail` in the script, only 2 of which would not need to set `status`. And I couldn't initialize `status` to invalid since it is used half-way through and would be set to OK there.... Overall, I felt the ugliness of this was outweighed by the clarity of having `goto fail`.